SwatDB/SwatDBDataObject.php

```diff
@@ -687,6 +710,48 @@ protected function getDeprecatedProperty($key)
 	}
 
 	// }}}
+	// {{{ protected function getProtectedPropertyList()
+
+	/**
+	 * Gets a list of all protected properties of this data-object
+	 *
+	 * The keys of this array should map to protected properties on this
+	 * object. The value of each entry is a second array in the format
+	 * [ 'get' => 'getMethodName', 'set' => 'setMethodName' ]
+	 * where the keys of each array correspond to methods on this object that
+	 * will be called by the magic getter and setter when the protected
+	 * property is accessed.
+	 *
+	 * @return array an array of protected property keys and method values.
+	 */
+	protected function getProtectedPropertyList()
+	{
+		return array();
+	}
+
+	// }}}
+	// {{{ protected function getProtectedProperties()
+
+	/**
+	 * Gets the protected properties of this data-object
+	 *
+	 * Protected properties should correspond directly to database fields.
+	 *
+	 * @return array a reference to an associative array of protected properties
+	 *                of this data-object. The array is of the form
+	 *                'property name' => 'property value'.
+	 */
+	protected function getProtectedProperties()
+	{
+		$properties = array();
+		foreach ($this->getProtectedPropertyList() as $property => $accessors) {
+			$properties[$property] = $this->$property;
```

Comment:
I am not completely sure if we want to use the accessor here. This method is used as part of part of the internal saving chain which means we probably don't want to return the values from the API here because then they will be saved into our DB.